WORKSPACE

```diff
@@ -611,18 +611,83 @@ load("@rules_jvm_external//:defs.bzl", "maven_install")
+        "com.google.flogger:flogger-system-backend:0.5.1",
+        "com.google.flogger:flogger:0.5.1",
+        "com.google.flogger:google-extensions:0.5.1",
+        "com.google.guava:failureaccess:1.0.1",
+        "com.google.guava:guava-testlib:31.1-jre",
+        "com.google.guava:guava:31.1-jre",
+        "com.google.http-client:google-http-client-gson:1.41.4",
+        "com.google.http-client:google-http-client:1.41.4",
+        "com.google.j2objc:j2objc-annotations:1.3",
+        "com.squareup:javapoet:1.8.0",
+        "commons-collections:commons-collections:3.2.2",
+        "commons-lang:commons-lang:2.6",
+        "io.github.java-diff-utils:java-diff-utils:4.0",
+        "io.grpc:grpc-api:1.47.0",
+        "io.grpc:grpc-auth:1.47.0",
+        "io.grpc:grpc-context:1.47.0",
+        "io.grpc:grpc-netty:1.47.0",
+        "io.grpc:grpc-protobuf-lite:1.47.0",
+        "io.grpc:grpc-protobuf:1.47.0",
+        "io.grpc:grpc-stub:1.47.0",
+        "io.netty:netty-all:4.1.87.Final",
+        "io.netty:netty-tcnative-boringssl-static:2.0.56.Final",
+        "io.netty:netty-tcnative-classes:2.0.56.Final",
+        "io.netty:netty-transport-native-unix-common:jar:linux-aarch_64:4.1.87.Final",
+        "io.netty:netty-transport-native-unix-common:jar:linux-x86_64:4.1.87.Final",
+        "io.netty:netty-transport-native-unix-common:jar:osx-aarch_64:4.1.87.Final",
+        "io.netty:netty-transport-native-unix-common:jar:osx-x86_64:4.1.87.Final",
+        "io.reactivex.rxjava3:rxjava:3.1.2",
+        "javax.activation:javax.activation-api:1.2.0",
+        "javax.annotation:javax.annotation-api:1.3.2",
+        "net.bytebuddy:byte-buddy-agent:1.9.7",
+        "net.bytebuddy:byte-buddy:1.9.7",
+        "org.apache.commons:commons-compress:1.19",
+        "org.apache.commons:commons-pool2:2.8.0",
+        "org.apache.tomcat:tomcat-annotations-api:8.0.5",
+        "org.apache.velocity:velocity:1.7",
+        "org.checkerframework:checker-qual:3.19.0",
+        "org.ow2.asm:asm-analysis:9.2",
+        "org.ow2.asm:asm-commons:9.2",
+        "org.ow2.asm:asm-tree:9.2",
+        "org.ow2.asm:asm-util:9.2",
+        "org.ow2.asm:asm:9.2",
+        "org.pcollections:pcollections:3.1.4",
+        "org.threeten:threeten-extra:1.5.0",
+        "org.tukaani:xz:1.9",
+    ],
+    excluded_artifacts = [
+        # org.apache.httpcomponents and org.eclipse.jgit:org.eclipse.jgit
+        # require java.security.jgss module to be embedded in the Bazel binary.
+        "org.apache.httpcomponents:httpclient",
+        "org.apache.httpcomponents:httpcore",
+        "org.eclipse.jgit:org.eclipse.jgit",
+        # We build protobuf Java library from source, exclude protobuf jars to be safe.
+        "com.google.protobuf:protobuf-java",
+        "com.google.protobuf:protobuf-javalite",
     ],
     repositories = [
-        "https://dl.google.com/android/maven2",
         "https://repo1.maven.org/maven2",
     ],
     maven_install_json = "//:maven_install.json",
```

Comment:
Consider using `strict_visibility` to make the transitive unspecified deps private: https://github.com/bazelbuild/rules_jvm_external#hiding-transitive-dependencies

Reply:
Awesome, this actually helped reduced the number of jars transitively introduced since I can remove netty-all and add back just the set of jars I need.